### PR TITLE
pretendard: init at 1.3.0

### PR DESCRIPTION
**Pull request by sudosubin**
###### Description of changes

`pretendard` is a system-ui alternative font for all cross-platform. It supports English, Korean, and Japanese. In `1.3.0`, it was differentiated into three typefaces (`Pretendard`, `PretendardJP`, `PretendardStd`).

https://github.com/orioncactus/pretendard

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2205.section.md [hunk 1]
```diff
@@ -141,6 +141,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - [kanidm](https://kanidm.github.io/kanidm/stable/), an identity management server written in Rust.
 
+- [pretendard](https://github.com/orioncactus/pretendard), a system-ui alternative font for all cross-platform.
```
Comment (dotlambda) [inline nixos/doc/manual/release-notes/rl-2205.section.md:144]: This is not a new NixOS module and doesn't warrent a release notes entry.

File: pkgs/data/fonts/pretendard/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      inherit sha256;
+
+      postFetch = ''
+        mkdir -p $out/share/fonts/opentype/${typeface}
+        unzip -j $downloadedFile "*.otf" -d $out/share/fonts/opentype/${typeface}/
+      '';
+
+      meta = with lib; {
+        homepage = "https://github.com/orioncactus/pretendard";
+        description = "Pretendard, a system-ui alternative font for all cross-platform.";
```
Comment (dotlambda) [inline pkgs/data/fonts/pretendard/default.nix:20]:
```suggestion
        description = "System-ui alternative font for all cross-platform";
```
See CONTRIBUTING.md.
However this is still not a proper English sentence.

File: pkgs/data/fonts/pretendard/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchzip }:
```
Comment (dotlambda) [inline pkgs/data/fonts/pretendard/default.nix:1]:
```suggestion
{ lib, fetchzip }:
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please add yourself to `maintainer-list.nix` in a separate commit.
It would also be nice to build the fonts from source.

File: pkgs/data/fonts/pretendard/default.nix (hunk 3 above)
Comment (sudosubin) [reply pkgs/data/fonts/pretendard/default.nix]: Thanks, I missed the unused inputs :)

File: pkgs/data/fonts/pretendard/default.nix (hunk 2 above)
Comment (sudosubin) [reply pkgs/data/fonts/pretendard/default.nix]: Thanks, now `meta.description` is `An alternative font to system-ui for all platforms`

**Comment by dotlambda** [comment]
> It would also be nice to build the fonts from source.

Have you given this a thought?

**Comment by sudosubin** [comment]
Pretendard is statically serving pre-built font files within the repository, version controlled. (same as GitHub Release assets)

In my opinion, when builing from source, there is a small chance that the font output will be slightly different from the actual Pretendard font being served.

So I expect it to be faster (package management, and package build from source), safe and accurate to use the font files provided as assets in the GitHub Release.

File: pkgs/data/fonts/pretendard/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  mkPretendard = { pname, typeface, sha256 }:
+    fetchzip {
+      name = "${pname}-${version}";
+
+      url = "https://github.com/orioncactus/pretendard/releases/download/v${version}/${typeface}-${version}.zip";
+      inherit sha256;
+
+      postFetch = ''
+        mkdir -p $out/share/fonts/opentype/${typeface}
+        unzip -j $downloadedFile "*.otf" -d $out/share/fonts/opentype/${typeface}/
```
[the remark this answers is not in the document]
Comment (sudosubin) [reply pkgs/data/fonts/pretendard/default.nix:15]: I think this is required to clearly distinguish the font path. (below output uses home-manager)

```sh
~/Library/Fonts/HomeManager/opentype
$ ls -al
total 0
drwxr-xr-x  May 17 00:33 .
drwxr-xr-x  May 17 00:33 ..
drwxr-xr-x  May 17 00:33 NerdFonts
drwxr-xr-x  May 17 00:33 Pretendard
drwxr-xr-x  May 17 00:33 PretendardJp
drwxr-xr-x  May 17 00:33 PretendardStd
drwxr-xr-x  May 17 00:33 noto-cjk
```

Is there any reason to unpack all font files in one path?

[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/data/fonts/pretendard/default.nix:15]: I don't think it's needed, most fonts don't do it.

[the remark this answers is not in the document]
Comment (sudosubin) [reply pkgs/data/fonts/pretendard/default.nix:15]: Oh I found many other fonts are really using that way. I will change to remove sub directory.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 173281` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>pretendard</li>
    <li>pretendard-jp</li>
    <li>pretendard-std</li>
  </ul>
</details>

**Pull request merged**